### DISCUSSION TITLE
Interop between Ant and different programming languages

### DISCUSSION THREAD

**Discussion by TheBlubb14** [AntMeNet/AntMeClassic#47] [Ideas]
Also one idea would be to define a [OpenAPI specification fka Swagger](https://github.com/OAI/OpenAPI-Specification) and then use [autorest tool](https://github.com/Azure/autorest) to generate client side code for different languages like C#, PowerShell, Go, Java, Node.js, TypeScript, Python.
We could then even provide a base language pack to the ant developers where they program with their ant and in the background the simulation server is called.
When having the simulation server talking over a web api with the ants, the server could validate the ant moves(anti cheat)

**Comment by tomwendel** [comment]
It's not that easy. We need a bidirectional (and fast) connection between simulation and implementation. For my current PoC I use Named Pipes and custom Messages. This needs to be implemented for all code runtimes to be compatible with the simulation process.

**Reply by TheBlubb14** [reply]
but for named pipes, the ant needs access to the filesystem?

**Reply by tomwendel** [reply]
Yes. Unsolved problem of secure the host system

**Reply by tomwendel** [reply]
For C# I can at least scan the IL code to make sure there is only a limited list of namespaces involved. But This does not work for other runtimes.
